server/internal/store/usage.go

```diff
@@ -0,0 +1,23 @@
+package store
+
+import "gorm.io/gorm"
+
+// Usage represents a API usage.
+type Usage struct {
```

Comment:
This is a question on the API design, but we could add some unique ID to `Usage` to eliminated duplicated craetion requests.

Reply:
I believe we can eliminate duplicates by creating a unique index using the combination of `user` and `timestamp` (unixnano), but I'm still wondering whether we should generate the IDs on each server. Is there anything I might be overlooking?

Reply:
Using user and timestamp sounds good!